specreduce/wavelength_calibration.py

```diff
@@ -0,0 +1,228 @@
+from astropy.modeling.models import Linear1D
+from astropy.modeling.fitting import LMLSQFitter, LinearLSQFitter
+from astropy.table import QTable, hstack
+import astropy.units as u
+from functools import cached_property
+from gwcs import wcs
+from gwcs import coordinate_frames as cf
+import numpy as np
+from specutils import Spectrum1D
+
+
+__all__ = ['WavelengthCalibration1D']
+
+
+def get_available_catalogs():
+    """
+    ToDo: Decide in what format to store calibration line catalogs (e.g., for lamps)
+          and write this function to determine the list of available catalog names.
+    """
+    return []
+
+
+def concatenate_catalogs():
+    """
+    ToDo: Code logic to combine the lines from multiple catalogs if needed
+    """
+    pass
+
+
+class WavelengthCalibration1D():
+
+    def __init__(self, input_spectrum, line_list, line_wavelengths=None, catalog=None,
+                 model=Linear1D(), spectral_unit=u.Angstrom, fitter=None):
+        """
+        input_spectrum: `~specutils.Spectrum1D`
+            A one-dimensional Spectrum1D calibration spectrum from an arc lamp or similar.
+        line_list: list, array, `~astropy.table.QTable`
+            List or array of line pixel locations to anchor the wavelength solution fit.
+            Will be converted to an astropy table internally if a list or array was input.
+            Can also be input as an `~astropy.table.QTable` table with (minimally) a column
+            named "pixel_center" and optionally a "wavelength" column with known line
+            wavelengths populated.
+        line_wavelengths: `~astropy.units.Quantity`, `~astropy.table.QTable`, optional
+            `astropy.units.Quantity` array of line wavelength values corresponding to the
+            line_pixels. Does not have to be in the same order (the lists will be sorted)
+            but does currently need to be the same length as line_list. Can also be input
+            as an `~astropy.table.QTable` with (minimally) a "wavelength" column.
+        catalog: list, str, optional
+            The name of a catalog of line wavelengths to load and use in automated and
+            template-matching line matching.
```

Comment:
could this be moved to a class method?  So `WavelengthCalibration1D.from_line_list(line_list)` calls `WavelengthCalibration1D(...)`?